src/Game.java

```diff
@@ -0,0 +1,67 @@
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static java.util.Map.Entry.comparingByValue;
+import static java.util.stream.Collectors.toMap;
+
+public class Game {
+
+     public void startGame(String requiredText) {
+        String strToGuess = createStringToGuess(requiredText.length());
+        Set<String> setForEnteredValues = new HashSet<String>();
+
+        while (strToGuess.contains("*")) {
+            Scanner scanner = new Scanner(System.in);
+            String userInput = scanner.nextLine().toLowerCase();
+
+            if (!isAlpha(userInput)) {
+                System.out.println("Please enter the correct symbol(s), only letters can be used in name");
+            }
+            if (setForEnteredValues.contains(userInput)) {
+                System.out.println("You already entered this string");
+            } else {
+                setForEnteredValues.add(userInput);
+                if (!requiredText.contains(userInput)) {
+                    System.out.println("Sorry but you didn't guess");
+                } else {
+                    while (requiredText.contains(userInput)) {
+                        int indexOfFinding = requiredText.indexOf(userInput);
+                        strToGuess = modifyStringToGuess(strToGuess, indexOfFinding, userInput);
+                        requiredText = modifyRequiredTextString(requiredText, indexOfFinding, userInput.length());
```

Comment:
good practice is not to modify your function arguments but create a modifiable copy if you need to modify smth